sync/class.jetpack-sync-actions.php

```diff
@@ -204,6 +199,25 @@ static function do_initial_sync( $new_version = null, $old_version = null ) {
 		self::do_full_sync( $initial_sync_config );
 	}
 
+	static function do_inital_sync_on_site_registration() {
+		$initial_sync_config = array(
+			'options'         => true,
+			'network_options' => true,
+			'functions'       => true,
+			'constants'       => true,
+		);
+
+		$user = wp_get_current_user();
+		if ( isset( $user->ID ) ) {
+			$initial_sync_config['users'] = array( $user->ID );
+		}
+
+		self::initialize_listener();
+		self::initialize_sender();
+		add_action( 'shutdown', array( self::$sender, 'do_full_sync' ) );
+		Jetpack_Sync_Modules::get_module( 'full-sync' )->start( $initial_sync_config );
```

Comment:
we should check if `Jetpack_Sync_Modules::get_module( 'full-sync' )` is available.

Reply:
How do we do that?
if( $full_sync = Jetpack_Sync_Modules::get_module( 'full-sync' ) ) ?

Reply:
before anything else;